components/host-dashboard/HostDashboardReports.js

```diff
@@ -179,28 +180,46 @@ const HostDashboardReports = ({ hostSlug }) => {
         </H1>
         <Box mx="auto" />
       </Flex>
-      <Flex flexWrap="wrap" mt="16px" mb="24px">
-        <Container width={[1, 1, 1 / 2]} pr={2} mb={[3, 3, 0, 0]}>
-          <FilterLabel htmlFor="transactions-period-filter">
-            <FormattedMessage id="TransactionsOverviewSection.PeriodFilter" defaultMessage="Filter by Date" />
-          </FilterLabel>
-          <PeriodFilter onChange={setDateInterval} value={dateInterval} minDate={host?.createdAt} />
-        </Container>
-        <Container width={[1, 1, 1 / 2]}>
-          <FilterLabel htmlFor="transactions-collective-filter">
-            <FormattedMessage id="TransactionsOverviewSection.CollectiveFilter" defaultMessage="Filter by Collective" />
-          </FilterLabel>
-          <CollectivePickerAsync
-            inputId="TransactionsCollectiveFilter"
-            data-cy="transactions-collective-filter"
-            types={[CollectiveType.COLLECTIVE, CollectiveType.EVENT, CollectiveType.PROJECT]}
-            isMulti
-            hostCollectiveIds={[host?.legacyId]}
-            onChange={value => setCollectives(prepareCollectivesForFilter(value))}
-            styles={{ control: baseStyles => ({ ...baseStyles, borderRadius: 16 }) }}
-          />
-        </Container>
-      </Flex>
+      <Container
+        position={['relative', 'sticky']}
+        top="0"
+        background="white"
+        zIndex="10"
+        py={3}
+        mb={4}
+        boxShadow="5px 12px 7px -10px #44444429"
+      >
+        <Grid gridTemplateColumns={['100%', '1fr 2fr auto']} gridGap={['8px', null, null, '28px']} width="100%">
+          <div>
+            <FilterLabel htmlFor="transactions-period-filter">
+              <FormattedMessage id="TransactionsOverviewSection.PeriodFilter" defaultMessage="Filter by Date" />
+            </FilterLabel>
+            <PeriodFilter onChange={setDateInterval} value={dateInterval} minDate={host?.createdAt} />
+          </div>
+          <div>
+            <FilterLabel htmlFor="transactions-collective-filter">
+              <FormattedMessage
+                id="TransactionsOverviewSection.CollectiveFilter"
+                defaultMessage="Filter by Collective"
+              />
+            </FilterLabel>
+            <CollectivePickerAsync
+              inputId="TransactionsCollectiveFilter"
+              data-cy="transactions-collective-filter"
+              types={[CollectiveType.COLLECTIVE, CollectiveType.EVENT, CollectiveType.PROJECT]}
+              isMulti
+              hostCollectiveIds={[host?.legacyId]}
+              onChange={value => setCollectives(prepareCollectivesForFilter(value))}
+              styles={{ control: baseStyles => ({ ...baseStyles, borderRadius: 16 }) }}
+              isLoading={loading}
+              disabled={loading}
```

Comment:
@Betree : Here probably we need to use the compact mode since otherwise the picker will not look good and grow 🤔